src/Handler/StreamHandler.php

```diff
@@ -114,7 +114,11 @@ private function createResponse(
         $status = (int) $parts[1];
         $reason = $parts[2] ?? null;
         $headers = Utils::headersFromLines($hdrs);
-        [$stream, $headers] = $this->checkDecode($options, $headers, $stream);
+
+        if (\strcasecmp('HEAD', $request->getMethod())) {
```

Comment:
As discussed previously this is not correct, but I'm using `strcasecmp` for consistency with the check below.